src/test/java/co/rsk/federate/FedNodeRunnerTest.java

```diff
@@ -74,11 +82,125 @@ public void setUp() throws IOException {
             mock(RskLogMonitor.class),
             mock(NodeRunner.class),
             fedNodeSystemProperties,
-            mock(HSMBookKeepingClientProvider.class),
+            hsmBookKeepingClientProvider,
             mock(FedNodeContext.class)
         );
     }
 
+    @Test
+    public void test_with_hsm_config_Ok() throws Exception {
+        SignerConfig btcSignerConfig = getHSMBTCSignerConfig();
+        SignerConfig rskSignerConfig = getHSMRSKSignerConfig();
+        SignerConfig mstSignerConfig = getHSMMSTSignerConfig();
+        when(fedNodeSystemProperties.signerConfig(BTC_KEY_ID.getId())).thenReturn(btcSignerConfig);
+        when(fedNodeSystemProperties.signerConfig(RSK_KEY_ID.getId())).thenReturn(rskSignerConfig);
+        when(fedNodeSystemProperties.signerConfig(MST_KEY_ID.getId())).thenReturn(mstSignerConfig);
+
+        fedNodeRunner.run();
+
+        ECDSASigner signer = TestUtils.getInternalState(fedNodeRunner, "signer");
+        assertNotNull(signer);
+        assertTrue(signer.canSignWith(BTC_KEY_ID.getKeyId()));
+        assertTrue(signer.canSignWith(RSK_KEY_ID.getKeyId()));
+        assertTrue(signer.canSignWith(MST_KEY_ID.getKeyId()));
+
+        assertTrue(signer instanceof ECDSACompositeSigner);
+        ECDSACompositeSigner compositeSigner = (ECDSACompositeSigner) signer;
+        List<ECDSASigner> signers = TestUtils.getInternalState(compositeSigner, "signers");
+        assertEquals(3, signers.size());
+        signers.forEach(hsmSigner -> assertTrue(hsmSigner instanceof ECDSAHSMSigner));
+
+        HSMBookkeepingClient hsmBookkeepingClient = TestUtils.getInternalState(fedNodeRunner, "hsmBookkeepingClient");
+        assertNotNull(hsmBookkeepingClient);
```

Comment:
Check `getVersion` returns the expected value?

Reply:
getVersion Was Mocked, It Will Be Redundant Asserting It